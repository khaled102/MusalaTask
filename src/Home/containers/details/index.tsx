import React, { memo, useEffect, useState } from 'react';
import {useSelector} from 'react-redux';
import {View, Image, Text} from 'react-native';
import styles from './style';
import moment from 'moment';
import {NewsCard} from '../../components';

type newsCardProps = {
  item: any;
  route: any;
};
export const NewsDetails = (props: newsCardProps) =>  {
  const { route } = props;
  const { item } = route.params;
  const themeData = useSelector((state: any) => state.setting.themeModeReducer);
  const [date, setDate] = useState<string>('');
  useEffect(() => {
    const newsDate = '2022-06-20T16:11:26Z';
    const formatedDate = moment(newsDate).format("MMM Do YY");
    setDate(formatedDate);
  }, []);
  return (
    <NewsCard 
      title={item.title}
      description={item.description}
      imageURL={item.urlToImage}
      author={item.author}
      publishedAt={date}
      theme={themeData.data}
      content={item.content}
      type={'fullScreen'}
    />
  );
}
