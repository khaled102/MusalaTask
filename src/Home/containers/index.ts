export * from './details';
export * from './newsFeed';
