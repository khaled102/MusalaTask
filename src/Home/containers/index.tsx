import React from 'react';
import {connect} from 'react-redux';
import {View, Text} from 'react-native';
import styles from './style';

export const HomeScreen = () =>  {
  return (
    <View style={styles.container}>
      <Text style={styles.helloWorld}>hello world</Text>
    </View>
  );
}
