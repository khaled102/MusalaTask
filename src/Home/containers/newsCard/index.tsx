import React, { useCallback, useEffect, useState } from 'react';
import {connect} from 'react-redux';
import {View, Image, Text} from 'react-native';
import styles from './style';
import moment from 'moment';

type newsCardProps = {
  title: string;
  description: string;
  imageURL: string;
};
export const NewsCard = (props: newsCardProps) =>  {
  const { title, description, imageURL } = props;
  const img = {uri: imageURL};
  const [date, setDate] = useState<String>('');
  useEffect(() => {
    const newsDate = '2022-06-20T16:11:26Z';
    const formatedDate = moment(newsDate).format("MMM Do YY");
    setDate(formatedDate);
  }, []);
  return (
    <View style={styles.container}>
      <Image 
        source={img}
        style={styles.image}
      />
      <Text style={styles.title}>
        {title}
      </Text>
      <Text style={styles.description}>{description}</Text>
      <View style={styles.footer}>
        <Text style={styles.footerText}>Created {date}</Text>
        <Text style={styles.footerText}>Author Kris Holt</Text>
      </View>
    </View>
  );
}
