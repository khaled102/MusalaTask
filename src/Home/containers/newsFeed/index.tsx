import React, { useCallback, useEffect, useState, useMemo } from 'react';
import {View, FlatList, RefreshControl} from 'react-native';
import styles from './style';
import { useSelector, useDispatch } from 'react-redux';
import {NewsCard} from '../../components';
import { news } from '../../actions';
import { SearchComponent } from '../../../Components';
import moment from 'moment';

export const HomeScreen = (props: any) =>  {
  const {navigation} = props;
  const dispatch = useDispatch();
  const newsData = useSelector((state: any) => state.home.newsReducer);
  const searchData = useSelector((state: any) => state.home.searchingReducer);
  const themeData = useSelector((state: any) => state.setting.themeModeReducer);
  const [refresh, setRefresh] = useState<boolean>(false);
  const [newsFeed, setNewsFeed] = useState<Array<any>>([]);
  const handleNavToDetails = (item: any) => {
    navigation.navigate('Details', {item: item});
  };
  const renderItem = ({item}: any) => {
    const newsDate = item.publishedAt;
    const formatedDate = moment(newsDate).format("MMM Do YY");
    return (
      <NewsCard 
        title={item.title}
        description={item.description}
        imageURL={item.urlToImage}
        author={item.author}
        publishedAt={formatedDate}
        theme={themeData.data}
        onPress={() => handleNavToDetails(item)}
      />
    )
  };
  const handleOnRefresh = useCallback(() => {
    dispatch(news());
  }, []);
  useEffect(() => {
    if (newsData.loading) {
      setRefresh(true);
    } else {
      setRefresh(false);
      setNewsFeed(newsData?.data?.articles);
    }
  }, [newsData]);
  useEffect(() => {
    if (searchData.data) {
      const Data: Array<any> = newsData?.data?.articles;
      let filter = Data.filter((item: any) => item?.title?.includes(searchData.data));
      if(filter.length === 0) {
        filter = Data.filter((item: any) => item?.description?.includes(searchData.data));
      }
      if(filter.length === 0) {
        filter = Data.filter((item: any) => item?.publishedAt?.includes(searchData.data));
      }
      if(filter.length === 0) {
        filter = Data.filter((item: any) => item?.author?.includes(searchData.data));
      }
      setNewsFeed(filter);
    } else if (!searchData.data) {
      setNewsFeed(newsData?.data?.articles);
    }
  }, [searchData]);
  return (
    <View style={styles.container}>
      <SearchComponent />
      {newsFeed &&
        <FlatList
          data={newsFeed} 
          renderItem={renderItem}
          directionalLockEnabled
          showsVerticalScrollIndicator={false}
          refreshControl={
            <RefreshControl 
              refreshing={refresh} 
              onRefresh={handleOnRefresh}
              tintColor="#F8852D"
            />
          }
        />
      }
    </View>
  );
}
