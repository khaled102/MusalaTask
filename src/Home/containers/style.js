import {StyleSheet} from 'react-native';

const styles = StyleSheet.create({
  container: {
    width: '100%',
    justifyContent: 'center',
    alignItems: 'center',
    marginTop: 100,
    backgroundColor: 'red',
  },
  helloWorld: {
    color: '#000000',
    fontSize: 16,
    alignSelf: 'center',
  },
});

export default styles;
