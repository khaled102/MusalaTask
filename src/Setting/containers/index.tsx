import React from 'react';
import {View, Text, TouchableOpacity} from 'react-native';
import styles from './style';

export const SettingScreen = () =>  {

  return (
    <View style={styles.container}>
      <TouchableOpacity onPress={handleChangeLanguage} style={styles.itemCard}>
        <Text style={styles.settingItem}>Change language</Text>
      </TouchableOpacity>
    </View>
  );
}
