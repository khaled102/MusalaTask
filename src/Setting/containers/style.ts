import {StyleSheet} from 'react-native';

const styles = StyleSheet.create({
  container: {
    flex: 1,
    width: '100%',
    alignItems: 'center',
    backgroundColor: '#FFFFFF',
  },
  itemCard: {
    width: '90%',
    backgroundColor: '#FFFFFF',
    borderBottomWidth: 0.2,
    borderColor: '#000',
    height: 80,
    alignItems: 'flex-start',
    justifyContent: 'center',
  },
  toggleCard: {
    width: '90%',
    flexDirection: 'row',
    backgroundColor: '#FFFFFF',
    borderBottomWidth: 0.2,
    borderColor: '#000',
    height: 80,
    alignItems: 'center',
    justifyContent: 'space-between',
  },
  settingItem: {
    color: '#000000',
    fontSize: 16,
    fontWeight: 'bold',
  },
});

export default styles;
