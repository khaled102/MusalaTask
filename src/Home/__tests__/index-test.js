import * as React from 'react';
import renderer from 'react-test-renderer';
import {HomeScreen} from '../containers';

it('renders correctly', () => {
  const homeRender = renderer.create(<HomeScreen />);
  expect(homeRender).toMatchSnapshot();
});
