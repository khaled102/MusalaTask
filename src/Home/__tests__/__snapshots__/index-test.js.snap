// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "red",
      "justifyContent": "center",
      "marginTop": 100,
      "width": "100%",
    }
  }
>
  <Text
    style={
      Object {
        "alignSelf": "center",
        "color": "#000000",
        "fontSize": 16,
      }
    }
  >
    hello world
  </Text>
</View>
`;
